src/js/angular/ttyg/directives/chat-view.directive.js

```diff
@@ -0,0 +1,179 @@
+
+            /**
+             * @type {ChatQuestion}
+             */
+            $scope.chatQuestion = undefined;
+
+            // =========================
+            // Private variables
+            // =========================
+
+            // =========================
+            // Public functions
+            // =========================
+
+            /**
+             * Handles the ask question action.
+             */
+            $scope.ask = () => {
+                if (!$scope.chatQuestion.conversationId) {
+                    createNewChat();
+                } else {
+                    askQuestion();
+                }
+            };
+
+            // =========================
+            // Private functions
+            // =========================
+
+            const setupNewChatQuestion = () => {
+                $scope.chatQuestion = getEmptyChatQuestion();
+            };
+
+            const createNewChat = () => {
+                TTYGContextService.emit(TTYGEventName.CREATE_CHAT, $scope.chatQuestion);
+            };
+
+            const askQuestion = () => {
+                if ($scope.selectedChat) {
+                    $scope.selectedChat.messages.push(chatQuestionToChatMessageMapper($scope.chatQuestion));
+                }
+                const question = cloneDeep($scope.chatQuestion);
+                setupNewChatQuestion();
+                TTYGContextService.emit(TTYGEventName.ASK_QUESTION, question);
+            };
+
+            /**
+             * Handles answering a chat question.
+             *
+             * @param {ChatMessageModel} answer
+             */
+            const onQuestionAnswer = (answer) => {
+                if ($scope.selectedChat && $scope.selectedChat.id === answer.conversationId) {
+                    $scope.selectedChat.messages.push(answer);
+                }
+            };
+
+            /**
+             * Handles the change of the selected chat.
+             * @param {ChatModel} chat - the new selected chat.
+             */
+            const onSelectedChatChanged = (chat) => {
+                if (!chat) {
+                    $scope.selectedChat = undefined;
+                    $scope.chatQuestion = getEmptyChatQuestion();
+                    return;
+                }
+                if ($scope.selectedChat && $scope.selectedChat.id === chat.id) {
+                    return;
+                }
+
+                // TODO: Check if the agent needs to be changed when the chat is switched.
+                TTYGService.getConversation(chat.id)
+                    .then((chat) => {
+                        $scope.selectedChat = chat;
+                        setupNewChatQuestion();
+                    });
+            };
+
+            /**
+             * Handles the change of the selected agent.
+             * @param {AgentModel} agent - the new selected agent.
+             */
+            const onSelectedAgentChanged = (agent) => {
```

Comment:
where is this used?

Reply:
This will be used once the agent functionality is ready. I've added a comment in the 'Subscriptions' section.